src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
If class MachineServiceImpl will be parameterized by Truck. Can we pass Bulldozer.class to the method getAll()? Is it valid behavior?
the same with other methods